Modules/Filtering/Smoothing/include/itkMeanImageFilter.hxx

```diff
@@ -82,11 +83,29 @@ MeanImageFilter<TInputImage, TOutputImage>::GenerateDataInSubregion(
     inputImage, Index<InputImageDimension>(), neighborhoodOffsets);
   auto outputIterator = ImageRegionRange<OutputImageType>(outputImage, imageRegion).begin();
 
+  InputRealType sumRef;
+  using PixelConvertType = DefaultConvertPixelTraits<InputRealType>;
+  using PixelComponentType = typename PixelConvertType::ComponentType;
+
+  PixelComponentType zeroComponent = NumericTraits<PixelComponentType>::ZeroValue(zeroComponent);
```

Comment:
This definition seems self-referential.

Reply:
I am not sure why/how it works, but I got this block of code from the ResampleImageFilter.

Reply:
Interesting PR, @blowekamp I'll have a look!

Reply:
Sorry, I won't be able to finish the review today.

Reply:
The line is no longer in the latest version.